test/hotspot/jtreg/serviceability/jvmti/vthread/PopFrameTest/PopFrameTest.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test id=default
+ * @summary Verifies JVMTI PopFrame support for virtual threads.
+ * @requires vm.continuations
+ * @run main/othervm/native -agentlib:PopFrameTest PopFrameTest
+ */
+
+/*
+ * @test id=no-vmcontinuations
+ * @summary Verifies JVMTI PopFrame support for bound virtual threads.
+ * @run main/othervm/native -agentlib:PopFrameTest -XX:+UnlockExperimentalVMOptions -XX:-VMContinuations PopFrameTest
+ */
+
+/*
+ * @test id=platform
+ * @summary Verifies JVMTI PopFrame support for platform threads.
+ * @run main/othervm/native -agentlib:PopFrameTest PopFrameTest platform
+ */
+
+import java.lang.AssertionError;
+
+/*
+ *     The test exercises the JVMTI function PopFrame.
+ *     The test creates a new virtual or platform thread.
+ *     Its method run() invokes the following methods:
+ *      - method A() that is blocked on a monitor
+ *      - method B() that is stopped at a breakpoint
+ *      - method C() that forces agent to call PopFrame on its own thread
+ *     JVMTI PopFrame is called in all cases.
+ */
+public class PopFrameTest {
+    private static final String agentLib = "PopFrameTest";
+    static final int JVMTI_ERROR_NONE = 0;
+    static final int THREAD_NOT_SUSPENDED = 13;
+    static final int OPAQUE_FRAME = 32;
+    static final int PASSED = 0;
+    static final int FAILED = 2;
+
+    static void log(String str) { System.out.println(str); }
```

Comment:
Better to flush system.out after each print.

Reply:
System.out has autoflush enabled so `println` will trigger a flush.